smart-contract/contracts/YourNftToken.sol

```diff
@@ -80,18 +80,20 @@ contract YourNftToken is ERC721A, Ownable, ReentrancyGuard {
     address latestOwnerAddress;
 
     while (ownedTokenIndex < ownerTokenCount && currentTokenId <= maxSupply) {
-      TokenOwnership memory ownership = _ownerships[currentTokenId];
+      if (_exists(currentTokenId))
+      {
+        TokenOwnership memory ownership = ownershipOf(currentTokenId);
 
-      if (!ownership.burned && ownership.addr != address(0)) {
-        latestOwnerAddress = ownership.addr;
-      }
+        if (!ownership.burned && ownership.addr != address(0)) {
```

Comment:
I think this `!ownership.burned` check can be omitted because it's implicit ([see `_exists(...)` implementation](https://github.com/chiru-labs/ERC721A/blob/95798665e86cf13d61ea6a1d585db87311149849/contracts/ERC721A.sol#L329)).

Reply:
@tomvanleeuwennl I think the change you made is not what I expected... you removed `latestOwnerAddress` completely, but that breaks the implementation, I was just talking about the `!ownership.burned` part in the IF statement. Am I missing anything?

Reply:
I tested the code locally and made some changes.